tests/dummy/app/routes/nodes/detail.js

```diff
@@ -3,28 +3,47 @@ import Ember from 'ember';
 // TODO: refactor permissions strings when https://github.com/CenterForOpenScience/ember-osf/pull/23/files#diff-7fd0bf247bef3c257e0fcfd7e544a338R5 is merged
 
 export default Ember.Route.extend({
-
     model(params) {
         return this.store.findRecord('node', params.node_id);
     },
 
     setupController(controller, model) {
         controller.set('editedTitle', model.get('title'));
+        controller.set('editedTitle', model.get('category'));
+        controller.set('editedTitle', model.get('dscription'));
         this._super(...arguments);
     },
 
     actions: {
-        editExisting(value) {
+        editExisting(title, description, category, isPublic) {
             // TODO: Should test PUT or PATCH
-            console.log('Will edit title from', this.modelFor(this.routeName).get('title'), ' to ', value);
+            // console.log('Will edit title from', this.modelFor(this.routeName).get('title'), ' to ', value);
             var node = this.modelFor(this.routeName);
             if (node.get('currentUserPermissions').indexOf('write') !== -1) {
-                node.set('title', value);
+                if (title) {node.set('title', title);}
+                if (category) {node.set('category', category);}
+                if (description) {node.set('description', description);}
+                if (isPublic !== null) {node.set('public', isPublic);}
                 node.save();
             } else {
                 console.log('You do not have permissions to edit this node');
             }
         },
+        affiliateNode(instId) {
+            var _this = this;
```

Comment:
If you use an ES6 arrow function in the promise, then `var _this = this;` should be unnecessary, and should be removed.